celery/utils/dispatch/signal.py

```diff
@@ -103,12 +109,43 @@ def connect(self, *args, **kwargs):
             dispatch_uid (Hashable): An identifier used to uniquely identify a
                 particular instance of a receiver.  This will usually be a
                 string, though it may be anything hashable.
+
+            retry (bool): If the signal receiver raises an exception
+                (e.g. ConnectionError), the receiver will be retried until it
+                runs successfully.
```

Comment:
@daviddavis Can a note be added here that identifies that `weak` is set to False whenever `retry == True` regardless of the value passed in to `connect()`.